src/OVAL/oval_enumerations.c

```diff
@@ -740,6 +740,13 @@ const char *oval_subtype_get_text(oval_subtype_t subtype)
 	}
 }
 
+bool oval_subtype_is_valid(oval_subtype_t subtype)
+{
+	const char *str = oval_subtype_get_text(subtype);
+	return (str != NULL && strcmp(str, _invalid) != 0);
```

Comment:
The function forces you have to resolve subtype text usually twice.

Why not to check only `char* subtype_text` ?